src/main/java/at/hannibal2/skyhanni/data/HypixelData.kt

```diff
@@ -485,32 +485,34 @@ object HypixelData {
     }
 
     private fun checkIsland(event: WidgetUpdateEvent) {
-        val islandType: IslandType
+        val newIsland: IslandType
         val foundIsland: String
         if (event.isClear()) {
 
             TabListData.fullyLoaded = false
-            islandType = IslandType.NONE
+            newIsland = IslandType.NONE
             foundIsland = ""
 
         } else {
             TabListData.fullyLoaded = true
             // Can not use color coding, because of the color effect (§f§lSKYB§6§lL§e§lOCK§A§L GUEST)
             val guesting = guestPattern.matches(ScoreboardData.objectiveTitle.removeColor())
             foundIsland = TabWidget.AREA.matchMatcherFirstLine { group("island").removeColor() } ?: ""
-            islandType = getIslandType(foundIsland, guesting)
+            newIsland = getIslandType(foundIsland, guesting)
         }
 
         // TODO don't send events when one of the arguments is none, at least when not on sb anymore
-        if (skyBlockIsland != islandType) {
-            IslandChangeEvent(islandType, skyBlockIsland).postAndCatch()
-            if (islandType == IslandType.UNKNOWN) {
+        if (skyBlockIsland != newIsland) {
+            val oldIsland = skyBlockIsland
+            skyBlockIsland = newIsland
```

Comment:
useful change, but also feels a bit out of scope of the pr

Reply:
this change is needed for this pr to work